### PR TITLE
Refactoring responses

### PR DESCRIPTION
**Pull request by silverAndroid**
This contains better error handling with a custom Error object, with the final definition of the new API responses. Due to the changes made to the tests in #27, the changes made here rely on it heavily and would not be backwards-compatible with what was on master before #27 was merged.

### CODE REVIEW

**Comment by silverAndroid** [comment]
This contains better error handling with a custom Error object, with the final definition of the new API responses. Due to the changes made to the tests in #27, the changes made here rely on it heavily and would not be backwards-compatible with what is currently on master

File: response-types.js [hunk 1]
```diff
@@ -1,27 +1,78 @@
...
+    super(title)
+    this.code = code
     this.title = title
+    this.status = status
+    Object.keys(extraProps).forEach(key => {
+      this[key] = extraProps[key]
+    })
+  }
+
+  static userDoesNotExist ({title = 'User doesn\'t exist', status = '404'} = {}) {
```
Comment (BrandonDanis) [inline response-types.js]: You should not be allowed to change the title according to http://jsonapi.org/format/#error-objects. We should add an optional message though, allowing to be a bit more specific about the issue

File: index.js [hunk 2]
```diff
@@ -18,6 +20,7 @@ app.get('/ping', (req, res) => {
 app.use('/users', require('./routes/user-routes'))
 app.use('/posts', require('./routes/post-routes'))
 app.use('/follow', require('./routes/follow-routes'))
+app.use(errorHandler)
```
Comment (BrandonDanis) [inline index.js]: 👍

File: response-types.js [hunk 3]
```diff
@@ -1,27 +1,79 @@
...
-  constructor (errors, meta = undefined) {
-    super(null, meta)
-    this.errors = errors
+// Only created like this for code completion
+let Response = {
+  OK: sendFakeResponse,
+  Created: sendFakeResponse,
+  Accepted: sendFakeResponse
+}
+Response = statuses.codes.filter(code => codes.some(usedCode => code === usedCode)).map(code => {
```
Comment (BrandonDanis) [inline response-types.js]: 🤢

Comment (silverAndroid) [reply response-types.js]: I don't see the problem here 😛

File: response-types.js [hunk 4]
```diff
@@ -1,27 +1,79 @@
...
+  // Generates each function
+  const functionName = statuses[code].replace(/[^a-zA-Z]/g, '')
+  if (code >= 400 && code < 600) {
```
Comment (BrandonDanis) [inline response-types.js]: Like we talked about, this can be removed 👍🏽

Comment (silverAndroid) [reply response-types.js]: Yep, the error functions will be removed

**Comment by BrandonDanis** [comment]
When we get a bad response, data is `null`
image

When we get a good response, error is not `null` but rather is `{}`.
image

Can we make it so error is `null` too please? 🙂

**Review by BrandonDanis — APPROVED** [review]
Although I haven't looked at all of it. If ever we run into a bug we can just fix it. No need to keep holding this off. LGTM 👍🏽

**Pull request merged**